libglusterfs/src/glusterfs/lkowner.h

```diff
@@ -19,8 +19,29 @@ lkowner_unparse(gf_lkowner_t *lkowner, char *buf, int buf_len)
 {
     int i = 0;
     int j = 0;
+    uint8_t nibble;
 
-    for (i = 0; i < lkowner->len; i++) {
+    /* the common case for lkowner is 8 bytes long
+       (a pointer) which will require 16 bytes (+ a NULL termination char)
+       long buffer to convert into.
+    */
+    if (lkowner->len >= 8 && buf_len > 16) {
+        for (i = 0; i < 8; i++) {
+            nibble = lkowner->data[i] & 0x0F;
+            if (nibble < 10)
+                buf[j] = '0' + nibble;
+            else
+                buf[j] = 'a' + nibble - 10;
+            j++;
+            nibble = (lkowner->data[i] & 0xF0) >> 4;
+            if (nibble < 10)
+                buf[j] = '0' + nibble;
+            else
+                buf[j] = 'a' + nibble - 10;
+            j++;
```

Comment:
The nibbles are written in reverse order. And I think a lookup table would be faster:
```suggestion
            buf[j++] = hex_digits[lkowner->data[i] >> 4];
            buf[j++] = hex_digits[lkowner->data[i] & 15];
```

Reply:
Fixed the nibble order. Unsure if I should have a table lookup (memory access) instead of an 'if' + simple assignment. Could be, but that goes beyond my nano optimizations already,